Source/Diagnostics/BTDiagnostics.H

```diff
@@ -14,47 +14,88 @@ private:
     /** Read relevant parameters for BTD */
     void ReadParameters ();
     /** Flush m_mf_output and particles to file. */
-    // Currently this is an empty function. When OpenPMD/Plotfile is supported
-    // this function can be moved to the base class Diagnostics.
-    void Flush (int /*i_buffer*/) override {}
+    // Currently, a temporary customized output format for the buffer
+    // data is implemented and called in this function.
+    // After implementing OpenPMD and plotfile formats, this function
+    // can be defined and implemented in Diagnostics.H, such that,
+    // the function call to flush out buffer data for
+    // FullDiagnostics and BTDiagnostics is the same
+    void Flush (int i_buffer) override;
     /** whether to write output files at this time step
      *  The data is flushed when the buffer is full and/or
-     *   when the simulation ends or when forced.
+     *  when the simulation ends or when forced.
      * \param[in] step current time step
      * \param[in] force_flush if true, return true for any step
      *  The return bool is used to determine when to write buffer data out
      */
     bool DoDump (int step, int i_buffer, bool force_flush=false) override;
+    /** whether to compute the back-transformed data and store buffer in this timestep
+     *  The field-data is back-transformed from boosted-frame to lab-frame
+     *  at every time step within the PIC loop. Back-transformation is not performed
+     *  at initialization.
+     * \param[in] step current time step, return true for any step >= 0
+     * \param[in] force_flush if true, return true for any step
+     */
     bool DoComputeAndPack (int step, bool force_flush=false) override;
     /** Initialize Data required to compute back-transformed diagnostics */
     void DerivedInitData () override;
     /** Initialize functors that store pointers to the fields requested by the user.
-     * \param[in] lev level on which the vector of unique_ptrs to field functors is initialized.
+     *  Additionally, the cell-center functors that stores pointers to all fields,
+     *  namely, Ex, Ey, Ez, Bx, By, Bz, jx, jy, jz, and rho is also initialized.
+     * \param[in] lev level on which the vector of unique_ptrs to field functors
+     *                is initialized.
      */
     void InitializeFieldFunctors (int lev) override;
-    /** This function initialized and allocates particle buffers for all the snapshots.
+    /** This function allocates and initializes particle buffers for all the snapshots.
      *  The particle containers required for this must be added to populate this function.
      */
```

Comment:
Do I understand this well?
```suggestion
    /** This function allocates and initializes particle buffers for all the snapshots.
     * This is currently an empty function:
     * The particle containers required for this must be added to populate this function.
     */
```

Reply:
YEs. it is correct that the function is currently empty and particle containers are required to populate this function